src/odemis/gui/cont/tabs.py

```diff
@@ -3305,7 +3287,7 @@ def _perform_switch_position_movement(self, target_button):
 
         # define the start position
         self._start_pos = current_pos = self._stage.position.value
-        current_position = getCurrentPositionLabel(current_pos, self._stage, self._aligner)
+        current_position = self.posture_manager.getCurrentPostureLabel()
 
         if self._role == 'enzel':
```

Comment:
I think that we could change the ENZEL code here to the same as the METEOR: it would use `self.posture_manager.getTargetPosition()`, instead of using its own dictionary. That would save a lot of code, as the dict is used just for this place.

We could also do that separately, as I think this code could be refactored even further...